tests/snuba/api/endpoints/test_organization_events_stats.py

```diff
@@ -563,6 +563,23 @@ def test_project_id_query_filter(self):
         )
         assert response.status_code == 200
 
+    def test_group_ids_filter(self):
+        # store transaction with group ids
+        _data = load_data("transaction")
+        # _data["group_ids"] = [123]
+        self.store_event(_data, self.project2.id)
+
+        response = self.do_request(
+            data={
+                "end": iso_format(before_now()),
+                "start": iso_format(before_now(hours=2)),
+                "query": "event.type:transaction performance.issue_ids:123",
```

Comment:
even when I specify `event.type:transaction` the query still searches in the `discover` dataset 🤔 
```
 {'app_id': 'default',
 'dataset': 'discover',
 'query': 'MATCH (discover) SELECT count() AS `count` BY time WHERE type = '
          "'transaction' AND ifNull(tags[performance.issue_ids], '') = '123' "
          "AND timestamp >= toDateTime('2022-08-30T18:56:22') AND timestamp < "
          "toDateTime('2022-08-30T20:56:22') AND project_id IN array(3, 2) "
          'ORDER BY time ASC LIMIT 10000 GRANULARITY 1800'}
api.organization-event-stats.sql:
 SELECT (toDateTime(multiply(intDiv(toUInt32(finish_ts), 1800), 1800), 'Universal') AS _snuba_time), (count() AS _snuba_count)
  FROM transactions_local
 WHERE equals(('transaction' AS _snuba_type), 'transaction')
   AND has(_tags_hash_map, cityHash64('performance.issue_ids=123'))
   AND greaterOrEquals((finish_ts AS _snuba_finish_ts), toDateTime('2022-08-30T18:56:22', 'Universal'))
   AND less(_snuba_finish_ts, toDateTime('2022-08-30T20:56:22', 'Universal'))
   AND in((project_id AS _snuba_project_id), [3, 2])
 GROUP BY _snuba_time
 ORDER BY _snuba_time ASC
 LIMIT 10000
OFFSET 0
```

@wmak What am I missing here?

Reply:
I believe you have to add group_ids [here in snuba](https://github.com/getsentry/snuba/blob/master/snuba/datasets/entities/discover.py#L247)

And I might have been wrong about [only needing group_ids on transactions, you may need it in discover too](https://github.com/getsentry/sentry/pull/38024/files#diff-1f77bf31e7ad04aa2f505c69d362ce4f087c18e439c3a769829506a2196472e5R36)

Reply:
ok, snuba PR here https://github.com/getsentry/snuba/pull/3104